src/api/catalogueItem.tsx

```diff
@@ -191,3 +194,56 @@ export const useEditCatalogueItem = (): UseMutationResult<
     }
   );
 };
+
+export const useMoveToCatalogueItem = (): UseMutationResult<
+  TransferState[],
+  AxiosError,
+  MoveToCatalogueItem
+> => {
+  const queryClient = useQueryClient();
+  return useMutation(async (moveToCatalogueItem: MoveToCatalogueItem) => {
+    const transferStates: TransferState[] = [];
+    let successfulIds: string[] = [];
+
+    const promises = moveToCatalogueItem.selectedItems.map(
+      async (item: EditCatalogueItem, index) => {
+        return editCatalogueItem({
+          id: item.id,
+          catalogue_category_id:
+            moveToCatalogueItem.targetCatalogueCategory?.id,
+        })
+          .then((result) => {
+            transferStates.push({
+              name: result.name ?? '',
+              message: `Successfully moved to ${
+                moveToCatalogueItem.targetCatalogueCategory?.name || 'Root'
+              }`,
+              state: 'success',
+            });
+            successfulIds.push(item.id);
+          })
+          .catch((error) => {
+            const response = error.response?.data as ErrorParsing;
+
+            transferStates.push({
+              name: moveToCatalogueItem.selectedItems[index].name,
+              message: response.detail,
+              state: 'error',
+            });
+          });
+      }
+    );
+
+    await Promise.all(promises);
+
+    if (successfulIds.length > 0) {
+      queryClient.invalidateQueries({ queryKey: ['CatalogueItems'] });
```

Comment:
Also just thought - Should we only invalidate the items needed here? We could do but not sure its worth the effort - the queries will refetch every so often anyway. I only did it for systems as it was simple to do due to the structure, but for catelogue categories it would require something like
![image](https://github.com/ral-facilities/inventory-management-system/assets/90245114/ceb6b5e8-1d18-4392-abc4-aeaaa9d21a6a)

Reply:
If you don't invalidate catalogue items the table is not going to update when the catalogue item is moved. I can change it so it only invalidate the queries for the target and the current location as those are the only only that need changing.

Reply:
![image](https://github.com/ral-facilities/inventory-management-system/assets/83226114/e191a1f0-3be5-47c9-9172-5f90e18f3d8c)